matrix/buffer.py

```diff
@@ -927,6 +927,13 @@ def __init__(self, room, server_name, homeserver, prev_batch):
             room.room_id
         )
 
+        if room.canonical_alias:
```

Comment:
This will get called only when the room is first created but a canonical alias can change or be set at a later point in time in the room history.

This should probably be moved into the `update_buffer_name()` method which gets called when we receive an event containing a canonical alias.

Reply:
Makes sense. I moved it. Please do tell if it needs moving again :+1: